frontend/app/services/redis-service.server.ts

```diff
@@ -21,52 +21,47 @@ import { type SetOptions, createClient } from 'redis';
 import { getEnv } from '~/utils/env.server';
 import { getLogger } from '~/utils/logging.server';
 
-const log = getLogger('redis-service.server');
-
-/**
- * Singleton redis client instance; lazy initialized as needed.
- */
-let redisClient: ReturnType<typeof createClient> | undefined = undefined;
-
-export function getRedisService() {
+function createRedisService() {
+  const log = getLogger('redis-service.server');
   const { REDIS_URL, REDIS_USERNAME, REDIS_PASSWORD } = getEnv();
 
-  function createRedisClient(url: string, username?: string, password?: string) {
-    log.info(`Creating new Redis client; url=[${url}]`);
-    return createClient({ url, username, password })
-      .on('connect', () => log.info(`Redis client initiating connection to [${url}]`))
+  // singleton redis client instance; lazy initialized as needed
+  let redisClient: ReturnType<typeof createClient> | undefined = undefined;
+
+  function createRedisClient() {
+    log.info(`Creating new Redis client; url=[${REDIS_URL}]`);
+    return createClient({ url: REDIS_URL, username: REDIS_USERNAME, password: REDIS_PASSWORD })
+      .on('connect', () => log.info(`Redis client initiating connection to [${REDIS_URL}]`))
       .on('ready', () => log.info('Redis client is ready to use'))
-      .on('reconnecting', () => log.info(`Redis client is reconnecting to [${url}]`))
-      .on('error', (error: Error) => log.error(`Redis client error connecting to [${url}]: ${error.message}`))
+      .on('reconnecting', () => log.info(`Redis client is reconnecting to [${REDIS_URL}]`))
+      .on('error', (error: Error) => log.error(`Redis client error connecting to [${REDIS_URL}]: ${error.message}`))
       .connect();
   }
 
-  async function getRedisClient() {
-    return (redisClient ??= await createRedisClient(REDIS_URL, REDIS_USERNAME, REDIS_PASSWORD));
-  }
-
   return {
     /**
      * @see https://redis.io/commands/get/
      */
     get: async (key: string) => {
-      const redisClient = await getRedisClient();
+      redisClient ??= await createRedisClient();
```

Comment:
I still thing the following function should be call instead of repeating this `redisClient ??= await createRedisClient();` each time you want to use the client.

Don't repeat yourself.
```
async function getRedisClient() {
    return (redisClient ??= await createRedisClient());
  }
```

or do the create and lazy loading in `getRedisClient()`.

Reply:
The logic around lazy loading is tricky. If you put it in the wrong place, it can accidentally init when you don't expect it to.

I'll move the init into `getRedisClient()`.

Reply:
Fixed in latest push.